### DISCUSSION TITLE
Planning on using Quarto by publishing to GitHub Pages, where does the desktop version come in?

### DISCUSSION THREAD

**Discussion by cyriltw** [quarto-dev/quarto-cli#7547] [Q&A]
### Description

Hi, 
I'm trying to understand how the whole ecosystem works. I'm planning on using Quarto to publish to GH Pages. I know there is an action already that can convert everything on the main branch to a GH Page. But there is a version to download that's on the website for the Mac and extension for VS Code. 

Something that's unclear to me based on the searching and docs is what role does the version that's downloaded for Mac play in the whole grand scheme of things? 

Side question: how do I update the Quarto version? seems like 1.4.x is on the horizon, but stable is 1.3.x. So when the new one is out, how do I update the website?

This is downloadable version -> https://quarto.org/docs/get-started/

Thanks!

**Comment by mcanouil** [comment]
The version you called "desktop" is not a desktop version.
Quarto is a command line interface tool (CLI).
The GitHub Action workflow use Quarto CLI to render your project and possibly publish it.
Even if some of the documentation to publish to GitHub Pages is a bit outdated, the fact that you first need to publish from your computer is still true.

Note that unless you intend to develop completely online without any preview before pushing to your main branch is a bit odd.

Usual workflow is to code and preview locally, then use CI/CD to publish/deploy the final version.
